sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLogSuite.scala

```diff
@@ -209,14 +209,21 @@ class HDFSMetadataLogSuite extends SparkFunSuite with SharedSQLContext {
     }
 
     // Open and delete
-    val f1 = fm.open(path)
-    fm.delete(path)
-    assert(!fm.exists(path))
-    intercept[IOException] {
-      fm.open(path)
+    // Open and delete
+    if (Utils.isWindows) {
+      Utils.tryWithResource(fm.open(path))(_ => ())
+      fm.delete(path)
+      fm.delete(path) // should not throw exception
+    } else {
+      Utils.tryWithResource(fm.open(path)) { _ =>
```

Comment:
We should close when it emits an exception. Otherwise, it is possible to suppress the actual assertion error due to the exception in removing the path, that makes the debug harder.